### PR TITLE
fix: Default callback is not called upon error

### PR DESCRIPTION
**Pull request by losalex**
Fixes #[<732>](https://github.com/googleapis/nodejs-logging-winston/issues/732) 🦕

### CODE REVIEW

File: src/common.ts [hunk 1]
```diff
@@ -312,7 +312,16 @@ export class LoggingCommon {
     // Make sure that both callbacks are called in case if provided
     const newCallback: Callback = (err: Error | null, apiResponse?: {}) => {
       if (callback) {
-        callback(err, apiResponse);
+        try {
+          callback(err, apiResponse);
+        } catch (err) {
```
[the remark this answers is not in the document]
Comment (losalex) [reply src/common.ts:317]: Thanks for a good catch - while apparently it will work, it is still good idea to use different names, so fixing it

**Comment by daniel-sanche** [comment]
We talked about this offline but to summarize for posterity: I have concerns that this catch block could hide errors. If `callback` throws an exception and `this.defaultCallback` is set, the error is swallowed. That could result in logs not making it into Cloud Logging, with no warnings being shown to the user. You told me that this should not be a concern because of how Winston callbacks are structured, so I'm going to try to look more into Winston before finishing the review

File: src/common.ts [hunk 2]
```diff
@@ -312,7 +312,17 @@ export class LoggingCommon {
...
       if (callback) {
-        callback(err, apiResponse);
+        try {
+          callback(err, apiResponse);
+        } catch (error) {
+          // We ignore here errors thrown from a callback calls (which is actually Writable.onwrite())
+          // since error could be temporary and following writes could succeed. We
+          // also want to make sure to call to defaultCallback suplied by a user, so
+          // user can get any details regarding error happened in defaultCallback
+          if (!this.defaultCallback) {
```
Comment (daniel-sanche) [inline src/common.ts:322]: One more potential solution

My main concern is around a specific edge case: where the callback throws a new exception, but `err` is still null, so nothing is shown to the user. From what you said, this should be impossible, but I'd feel better if we still added a guard against the impossible case, just to be safe.

What if we changed this function to something like this:

```
const newCallback: Callback = (err: Error | null, apiResponse?: {}) => {
      var callback_error = null
      if (callback) {
        try {
          callback(err, apiResponse);
        } catch (error) {
          callback_error = error
        }
      }
      if (this.defaultCallback) {
        this.defaultCallback(err, apiResponse);
      }
      if (callback_error && (!this.defaultCallback || err==null)){
          // we want to raise an error in two cases: if the defaultCallback isn't set, 
          // or if the callback throws a different error than the original call
          throw callback_error
      }
};
```


`if (callback_error && !this.defaultCallback)` should be equivalent to the existing logic. `if (callback_error && error==null)` shouldn't happen based on what you told me, but it makes sure no exceptions go unreported if our assumptions are broken

What do you think?

Comment (losalex) [reply src/common.ts]: Seems I came to same conclusion as you and worked on same idea for a fix :-) But I agree, this is a better way to handle the case.

**Review by daniel-sanche — APPROVED** [review]

**Pull request merged**